### PR TITLE
Update changelog for PRs to 4.1

### PR DESCRIPTION
**Pull request by jcape**
- Update 4.1.0 changelog
- Fix links in 4.0.2, 4.0.1 changelogs.

### Motivation

Pre-requisite to releasing 4.1.

### CODE REVIEW

**Comment by jcape** [comment]
Current dependencies on/for this PR:
* release/v4.1
  * **PR #3162** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3162" target="_blank">Graphite</a> 
    * **PR #3246** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3246" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3246?utm_source=stack-comment).

File: CHANGELOG.md [hunk 1]
```diff
@@ -230,7 +256,6 @@ The crates in this repository do not adhere to [Semantic Versioning](https://sem
 [#2779]: https://github.com/mobilecoinfoundation/mobilecoin/pull/2779
 [#2776]: https://github.com/mobilecoinfoundation/mobilecoin/pull/2776
 [#2777]: https://github.com/mobilecoinfoundation/mobilecoin/pull/2777
```
Comment (nick-mobilecoin) [inline CHANGELOG.md:258]: I don't think 2575 should have gone away it's referenced on 122
```suggestion
[#2777]: https://github.com/mobilecoinfoundation/mobilecoin/pull/2777
[#2575]: https://github.com/mobilecoinfoundation/mobilecoin/pull/2575
```

**Review by nick-mobilecoin — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +51,10 @@ The crates in this repository do not adhere to [Semantic Versioning](https://sem
 
 - Improved performance of mobilecoind-dev-faucet for better load testing ([#2955])
 
+[#2988]: https://github.com/mobilecoinfoundation/mobilecoin/pull/2988
+[#2952]: https://github.com/mobilecoinfoundation/mobilecoin/pull/2952
+[#2955]: https://github.com/mobilecoinfoundation/mobilecoin/pull/2955
```
Comment (wjuan-mob) [inline CHANGELOG.md:56]:
```suggestion
[#3062]: https://github.com/mobilecoinfoundation/mobilecoin/pull/3062
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -29,6 +51,10 @@ The crates in this repository do not adhere to [Semantic Versioning](https://sem
...
+
```
Comment (wjuan-mob) [inline CHANGELOG.md:57]:
```suggestion
[#3062]: https://github.com/mobilecoinfoundation/mobilecoin/pull/3062

```

Comment (jcape) [reply CHANGELOG.md:57]: Where is this referenced?

**Review by wjuan-mob — COMMENTED** [review]
Should we mention https://github.com/mobilecoinfoundation/mobilecoin/pull/2926 somewhere in here? I'm not sure what line items make it into the changelog.

File: CHANGELOG.md (hunk 3 above)
Comment (wjuan-mob) [reply CHANGELOG.md:57]: Sorry. I'm not sure what you mean. I thought it should be included because it was part of the dev faucet changes.

**Comment by jcape** [comment]
Ugh, CI won't run because this is "just a markdown change", so the required tests will never pass. Once it has reviews I'm going to force merge it.

**Review by wjuan-mob — APPROVED** [review]

**Pull request merged**